File: ios/Sources/AppDiagLog/Storage/ExportManifest.swift
import Foundation

/// Plaintext manifest bundled into the export zip.
struct ExportManifest: Codable, Sendable {
    let version: Int
    let sdkVersion: String
    let exportedAt: String
    let sessions: [Session]

    struct Session: Codable, Sendable {
        let id: String
        let createdAt: String
        let sealedAt: String?
        let eventCount: Int
        let sessionTag: String?
        let fileName: String

        enum CodingKeys: String, CodingKey {
            case id
            case createdAt = "created_at"
            case sealedAt = "sealed_at"
            case eventCount = "event_count"
            case sessionTag = "session_tag"
            case fileName = "file_name"
        }
    }

    enum CodingKeys: String, CodingKey {
        case version, sessions
        case sdkVersion = "sdk_version"
        case exportedAt = "exported_at"
    }
}

File: backend/src/main/kotlin/com/appdiaglog/server/model/ExportManifest.kt
package com.appdiaglog.server.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Plaintext manifest written at the root of every export ZIP. Lists which
 * `sessions/x.enc` files are present and their plaintext metadata.
 *
 * Useful before decryption — you can triage by app version / device model
 * without touching key material.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
data class ExportManifest(
    val version: Int,
    @JsonProperty("sdk_version") val sdkVersion: String,
    @JsonProperty("exported_at") val exportedAt: String,
    val sessions: List<Session>,
) {
    @JsonInclude(JsonInclude.Include.NON_NULL)
    data class Session(
        val id: String,
        @JsonProperty("created_at") val createdAt: String,
        @JsonProperty("sealed_at") val sealedAt: String? = null,
        @JsonProperty("event_count") val eventCount: Int,
        @JsonProperty("session_tag") val sessionTag: String? = null,
        @JsonProperty("file_name") val fileName: String,
    )
}
